File: sources/py/1.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        for i in range(len(nums)):
            x = target - nums[i] # 求与目标值的差值
            if nums[i+1:].count(x) >= 1: # 如果后面存在
                return [i, nums[i+1:].index(x)+i+1]
            elif nums[:i].count(x) >= 1: # 如果前面存在
                return [nums.index(x), i]

File: sources/java/1.java
public class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] arr = { -1, -1 };
        if (nums != null && nums.length >= 2) {
            HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
            for (int i = 0; i < nums.length; i++) {
                if (map.containsKey(target - nums[i])) {
                    arr[0] = map.get(target - nums[i]);
                    arr[1] = i;
                    break;
                }
                map.put(nums[i], i);
            }
        }
        return arr;
    }
}
